vavr/src/main/java/io/vavr/control/Option.java

```diff
@@ -361,8 +371,51 @@ default T getOrElse(Supplier<? extends T> supplier) {
     }
 
     /**
-     * Maps the value and wraps it in a new {@code Some} if this is a {@code Some}, returns {@code None}.
+     * Maps the value to a new {@code Option} if this is a {@code Some}, otherwise returns {@code None}.
+     * In the case the given {@code mapper} throws, {@code None} will be returned.
+     * <p>
+     * Examples:
+     * <pre>{@code
+     * // = None
+     * Option.<Integer> none().flatMapTry(i -> 1 / i);
+     *
+     * // = Some(0)
+     * Option.some(2).flatMapTry(i -> Option.of(1 / i));
      *
+     * // = None
+     * Option.some(0).flatMapTry(i -> Option.of(1 / i)); // division by zero
+     * }</pre>
+     *
+     * @param mapper A value mapper that may throw
+     * @param <U>    Component type of the resulting Option
+     * @return a new {@code Option}
+     */
+    @SuppressWarnings("unchecked")
+    default <U> Option<U> flatMapTry(Function<? super T, ? extends Option<? extends U>> mapper) {
+        Objects.requireNonNull(mapper, "mapper is null");
+        if (isEmpty()) {
+            return none();
+        } else {
+            try {
+                return (Option<U>) mapper.apply(get());
+            } catch(Throwable x) {
```

Comment:
catching `Throwable` is generally bad form as this can be Errors like `OutOfMemoryError` or `ExceptionInInitializerError`. These are considered non-recoverable and it's explicitly encouraged to exit the JVM. How about `catch(Exception e) ` as this covers every conceivable non-error case.

Reply:
That is a valid point! We need some kind of re-throw logic in the fatal case that will be re-added when we have solved #2049. See also https://github.com/vavr-io/vavr/blob/master/vavr/src/main/java/io/vavr/control/Try.java#L1522-L1527